### PR TITLE
Feature - Edit Profile Component

### PR DESCRIPTION
**Pull request by jv-farias**
### Preview Storybook

https://github.com/dcTeam23/cat-list/assets/84153149/909accfd-ef6e-4c8b-82f3-a1c319e5043f

### CODE REVIEW

File: src/components/EditProfile/EditProfile.jsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import { Label } from "../ui/label";
+import { useState } from "react";
+
+export const EditProfile = () => {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [newPassword, setNewPassword] = useState("");
+
+  const handleInputChange = (/** @type {{ (value: import("react").SetStateAction<string>): void;}} */ setter) => (/** @type {{ target: { value: string; }; }} */ e) => {
```
Comment (AbraaoAlves) [inline src/components/EditProfile/EditProfile.jsx:19]:
``` suggestion
  /**
   * @param { { (value: import("react").SetStateAction<string>): void; }} setter
   * @returns { (e: import("react").ChangeEvent<HTMLInputElement>) => void }
   */
  const handleInputChange = (setter) => (e) => {
```

File: src/components/EditProfile/EditProfile.jsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    setter(e.target.value);
+  };
+
+  const handleSubmitName = (/** @type {{ preventDefault: () => void; }} */ e) => {
```
Comment (AbraaoAlves) [inline src/components/EditProfile/EditProfile.jsx:23]:
``` suggestion
  /**
   * @param {import("react").MouseEvent} e
   */
  const handleSubmitName = (e) => {
```

File: src/components/EditProfile/EditProfile.jsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    e.preventDefault();
+    name && email ? console.log("Name and Email saved") : null;
+    setName("");
+    setEmail("");
+  };
+
+  const handleSubmitPassword = (/** @type {{ preventDefault: () => void; }} */ e) => {
```
Comment (AbraaoAlves) [inline src/components/EditProfile/EditProfile.jsx:30]:
``` suggestion
  /**
   * @param {import("react").MouseEvent} e
   */
  const handleSubmitPassword = (e) => {
```

**Review by AbraaoAlves — COMMENTED** [review]
try keep jsdoc comments on top of the functions

**Comment by jv-farias** [comment]
thank you very much for your help @AbraaoAlves. in the next features we will use this way

**Comment by RodrigoNet0** [comment]
ok,thank you very  much Mr.

**Pull request merged**